subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoReportAggregationPlugin.java

```diff
@@ -63,35 +60,34 @@ public abstract class JacocoReportAggregationPlugin implements Plugin<Project> {
 
     @Override
     public void apply(Project project) {
-        project.getPluginManager().apply("org.gradle.reporting-base");
         project.getPluginManager().apply("jvm-ecosystem");
         project.getPluginManager().apply("jacoco");
 
+
         RoleBasedConfigurationContainerInternal configurations = ((ProjectInternal) project).getConfigurations();
         Configuration jacocoAggregation = configurations.bucket(JACOCO_AGGREGATION_CONFIGURATION_NAME);
         jacocoAggregation.setDescription("Collects project dependencies for purposes of JaCoCo coverage report aggregation");
         jacocoAggregation.setVisible(false);
-        jacocoAggregation.setTransitive(true);
 
-        ObjectFactory objects = project.getObjects();
-        @SuppressWarnings("deprecation") Configuration codeCoverageResultsConf = configurations.createWithRole("aggregateCodeCoverageReportResults", ConfigurationRolesForMigration.RESOLVABLE_BUCKET_TO_RESOLVABLE);
-        codeCoverageResultsConf.setDescription("Graph needed for the aggregated JaCoCo coverage report.");
+        Configuration codeCoverageResultsConf = configurations.createWithRole("aggregateCodeCoverageReportResults", ConfigurationRoles.RESOLVABLE);
+        codeCoverageResultsConf.setDescription("Resolvable configuration used to gather files for the JaCoCo coverage report aggregation via ArtifactViews, not intended to be used directly");
         codeCoverageResultsConf.extendsFrom(jacocoAggregation);
         codeCoverageResultsConf.setVisible(false);
-        jacocoAggregation.setTransitive(true);
+
+        project.getPlugins().withType(JavaBasePlugin.class, plugin -> {
+            // If the current project is jvm-based, aggregate dependent projects as jvm-based as well.
+            getEcosystemUtilities().configureAsRuntimeClasspath(codeCoverageResultsConf);
+        });
+
+        ObjectFactory objects = project.getObjects();
 
         ArtifactView sourceDirectories = codeCoverageResultsConf.getIncoming().artifactView(view -> {
             view.withVariantReselection();
             view.componentFilter(id -> id instanceof ProjectComponentIdentifier);
-            view.attributes(attributes -> {
-                attributes.attribute(Bundling.BUNDLING_ATTRIBUTE, objects.named(Bundling.class, Bundling.EXTERNAL));
-                attributes.attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category.class, Category.VERIFICATION));
-                attributes.attribute(VerificationType.VERIFICATION_TYPE_ATTRIBUTE, objects.named(VerificationType.class, VerificationType.MAIN_SOURCES));
-            });
+            getEcosystemUtilities().configureAsSources(view);
```

Comment:
I'm not sure about this. 

If anything, we should get rid of JvmEcosystemUtilities because it makes something simple (a list of attributes) really complicated (builders).  It also conflates the attributes for a variant with the attributes for a query. These will almost always be different things with the variant having more attributes than the query. 

Let's merge this as-is.

Reply:
By "as-is" do you mean leave this change even though you're not sure of it or do you mean revert this change before merging.

I agree about the builders being more trouble that they're worth, but I do think we need a method to group related sets of attributes for application on the producer and consumer sides.  I think these could be named in such a way to not suggest these add the _complete_ set of attributes, but merely the relevant subset that powers the matching.